freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -184,11 +187,9 @@ private void logAffectedStacks(List<Stack> stacks) {
         LOGGER.info("Affected stacks: {}", stacksAffected);
     }
 
-    private List<Stack> getStacksForSync(String accountId, String actorCrn, Set<String> environmentCrnFilter,
-            Set<String> userCrnFilter, Set<String> machineUserCrnFilter) {
-        validateParameters(accountId, actorCrn, environmentCrnFilter, userCrnFilter, machineUserCrnFilter);
-        LOGGER.debug("Synchronizing users in account {} for environmentCrns {}, userCrns {}, and machineUserCrns {}",
-                accountId, environmentCrnFilter, userCrnFilter, machineUserCrnFilter);
+    private List<Stack> getStacksForSync(String accountId, String actorCrn, Set<String> environmentCrnFilter, UserSyncRequestFilter useSyncFilter) {
+        validateParameters(accountId, actorCrn, environmentCrnFilter, useSyncFilter.getUserCrnFilter(), useSyncFilter.getMachineUserCrnFilter());
```

Comment:
validateParameters should be updated for your userSyncRequestFilter including the users+machineUsers size when a deletedWorkloadUsername is included

Reply:
Ah, this check was in the constructor, but I can move it to here instead.